scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -246,8 +263,129 @@ public SchedulerDBManager(Configuration configuration, boolean drop) {
+            final String user, final boolean pending, final boolean running,
+            final boolean finished, final List<SortParameter<TaskSortParameter>> sortParameters) {
+
+        DBTaskDataParameters parameters = new DBTaskDataParameters(tag, from, to, offset, limit, user,
+                pending, running, finished, sortParameters);
+        int totalNbTasks = getTotalNumberOfTasks(parameters);
+        List<TaskInfo> lTaskInfo =  runWithoutTransaction(TaskDBUtils.taskInfoSessionWork(parameters));
+        
+        return new Page<TaskInfo>(lTaskInfo, totalNbTasks);
+    }
+    
+    private int getTotalNumberOfTasks(final DBTaskDataParameters params) {
+
+        return runWithoutTransaction(new SessionWork<Integer>() {
+
+            @Override
+            public Integer executeWork(Session session) {
+
+                Set<TaskStatus> lStatuses = params.getStatuses();
+
+                if (lStatuses.isEmpty()) {
+                    return 0;
+                } else {
+
+                    boolean hasUser = params.getUser() != null && "".compareTo(params.getUser()) != 0;
+                    boolean hasTag = params.getTag() != null && "".compareTo(params.getTag()) != 0;
+                    boolean hasDateFrom = params.getFrom() != 0;
+                    boolean hasDateTo = params.getTo() != 0;
+
+                    StringBuilder queryString = new StringBuilder(
+                        "select count(*) from TaskData T where T.jobData.removedTime = -1 ");
+
+                    if (hasDateFrom)
+                        queryString.append("and startTime >= :dateFrom ");
+
+                    if (hasDateTo)
+                        queryString.append("and finishedTime <= :dateTo ");
+
+                    if (hasUser)
+                        queryString.append("and T.jobData.owner = :user ");
+
+                    if (hasTag)
+                        queryString.append("and tag = :taskTag ");
+
+                    queryString.append("and taskStatus in (:taskStatus) ");
+
+                    Query query = session.createQuery(queryString.toString());
+                    query.setParameterList("taskStatus", lStatuses);
+                    if (hasUser) {
+                        query.setParameter("user", params.getUser());
+                    }
+
+                    if (hasDateFrom)
+                        query.setParameter("dateFrom", params.getFrom());
+                    if (hasDateTo)
+                        query.setParameter("dateTo", params.getTo());
+                    if (hasTag)
+                        query.setParameter("taskTag", params.getTag());
+                    Long count = (Long) query.uniqueResult();
+                    
+                    return count.intValue();
+                }
+            }
+        });
     }
+    
+    private int getTotalNumberOfJobs(final DBJobDataParameters params) {
+
+        return runWithoutTransaction(new SessionWork<Integer>() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 33 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1188)